src/librustc_typeck/check/mod.rs

```diff
@@ -4453,16 +4453,6 @@ impl<'a, 'gcx, 'tcx> FnCtxt<'a, 'gcx, 'tcx> {
                     }
                 };
 
-                if let Ok(count) = count {
-                    let zero_or_one = count.assert_usize(tcx).map_or(false, |count| count <= 1);
-                    if !zero_or_one {
```

Comment:
Presumably we'll want to feature gate this?

Reply:
We can feature-gate the promotion.

Reply:
I've added a feature gate and use it in `qualif_consts` (I'm not sure if you wanted it to be used in `promote_consts` or `qualif_consts` but it was slightly easier add to `qualif_consts`).